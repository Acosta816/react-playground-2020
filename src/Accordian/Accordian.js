import React from 'react';


class Accordian extends React.Component {

    static defaultProps = {
        sections: []
    }

    state = {
        currentTab: 0
    }

    renderButtons = ()=> {
        const myButtons = this.props.sections.map((section,index)=> {
            return(
                <li key={index}>
                    <button onClick={()=>{this.changeState(index)}} >{section.name}</button>
                    { index === this.state.currentTab ? this.renderContent(index) : null }
                </li>
            )
        })

        return myButtons;
    }

    renderContent = (index)=> {
        return (
            <div>
                <p>{this.props.sections[index].content}</p>
                <img src={this.props.sections[index].avatar} alt="sonic gifs" />
            </div>
        )
    }

    changeState = (index)=> {
        console.log(index);
        this.setState({
            currentTab: index
        })
    }

    render() {

        return(
            <div>
                <ul>
                    {this.props.sections.length ? this.renderButtons() : null }
                </ul>
            </div>
        )
    }
}


export default Accordian;