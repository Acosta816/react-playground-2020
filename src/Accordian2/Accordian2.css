.current-content img {
    width: 250px;
    height: auto;
}