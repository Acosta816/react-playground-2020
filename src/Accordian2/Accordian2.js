import React from 'react';
import './Accordian2.css';

class Accordian2 extends React.Component {

    static defaultProps = {
        sections: [{name:"NO SECTIONS FOUND", content: "", avatar: ""}],
        currentSection: {name:"NO SECTIONS FOUND", content: "", avatar: ""}
    }

    constructor(props){
        super(props);
        this.state = {
            sections: [],
            currentSection: this.props.sections[0]
        }
    }

    componentDidMount() {
        this.setState({
            sections: this.props.sections,
        })
    }

    renderContent = ()=> {
        const {currentSection} = this.state;
        return(
            <div className="current-content">
                <img src={currentSection.avatar} alt="content avatar"/>
                <h2>{currentSection.content}</h2>
            </div>
        )
    }

    handleButtonClick = (e)=> {
        const currentSection = this.state.sections.find(section=> section.name === e.target.innerText);
        console.log(currentSection);
        this.setState({
            currentSection
        })
    }



    renderSections = ()=> {
        const sectionsArray = this.props.sections.map((section,index)=> {
            return(
                <li key={index}>
                    <button type="button"  onClick={(e)=> {this.handleButtonClick(e)}}>{section.name}</button>
                    {this.state.currentSection.name === section.name ? this.renderContent() : null}
                </li>
            )
        })

        return sectionsArray;
    }

    render(){
        console.log(this.state)
        return(
            <div>
                {this.renderSections()}
            </div>
        )
    }
}

export default Accordian2;