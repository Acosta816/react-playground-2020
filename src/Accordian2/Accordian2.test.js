import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian2 from './Accordian2';
import Accordian from '../Accordian/Accordian';


describe('TestSuite: Accordian2 component:', ()=> {
    const sections = [{name:'Sonic', content: 'Fastest thing Alive', avatar: "https://media2.giphy.com/media/K3B3v1IUkroE8/giphy.gif?cid=790b7611a0141074eddd3986bac2408e78aa2c9921709e38&rid=giphy.gif"}, {name:'Tails', content: 'Loyal and Hyper-intelligent Flying Fox', avatar: "https://66.media.tumblr.com/a2b506d774c4840d65527670dd77076e/tumblr_pc8gylOts11qjztgpo1_400.gifv"}, {name:'Amy', content: 'Sonics Biggest Fan. Hammer-weilding Pink Hedgehog', avatar: 'https://appstickers-cdn.appadvice.com/454316134/827794092/56d7cab123afdb67c732b5ceb21206ef-1.gif'}, {name:'Robotnik', content: 'Crazed Genius, obsessed with robotisizing all life and ridding the world of Sonic', avatar: 'https://thumbs.gfycat.com/RepentantPepperyBangeltiger.webp'}, {name:'Shadow', content: 'Sonics adversary. Created by Robotnik to become the ultimate lifeform', avatar: "https://pa1.narvii.com/6495/a59913f75290d830cfb61f2651917c6b2ce3400b_hq.gif"}, {name:'Sally', content: 'Rightful Princess of Mobius and one of the Leaders in the Freedom Fighters against Robotnik. Sonics Best friend.', avatar: "https://thumbs.gfycat.com/TiredFlakyBlesbok-max-1mb.gif"}];

    it('SMOKE: renders without crashing', ()=> {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Snap1: renders empty with no sections prop', ()=> {
        const wrapper = shallow(<Accordian2 />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('Snap2: closes first section and opens third section', ()=> {
        const wrapper = shallow(<Accordian2 sections={sections}/>);
        console.log(wrapper.find('button').at(2).debug());
        wrapper.find('button').at(2).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    })

})