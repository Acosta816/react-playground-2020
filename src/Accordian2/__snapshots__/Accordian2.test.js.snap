// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite: Accordian2 component: Snap1: renders empty with no sections prop 1`] = `
<div>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      NO SECTIONS FOUND
    </button>
    <div
      className="current-content"
    >
      <img
        alt="content avatar"
        src=""
      />
      <h2 />
    </div>
  </li>
</div>
`;
